lib/blockchain.py

```diff
@@ -176,12 +176,18 @@ def choose_transactions_from_mempool(self) -> List[Transaction]:
 
       chosen_transaction = self.choose_random_transaction(transaction_bucket)
       sender_account, recipient_account = self.get_transaction_accounts(chosen_transaction)
-      if recipient_account.is_contract():
-        gas = recipient_account.calc_gas()
-        if sender_account.charge_gas(gas):
-          recipient_account.exec_contract(chosen_transaction)
-          self.update_accounts(sender_account)
-          chosen_transactions.append(chosen_transaction)
+      if sender_account.is_contract():
+        # Retrieve the function name from the data
+        func_name = chosen_transaction.get_data()['func_name']
+        # Retreive the function arguments from the data
+        func_args = chosen_transaction.get_data()['func_args']
+        # Check if they are the correct type
+        if type(func_name) is str and type(func_args) is tuple:
```

Comment:
btw why do u have to check the type again if you're setting them from the opcode?

Reply:
So i'm not setting it from opcode. I'm using the name of the function to execute the function. Let me fix up the account. It'll be more clear then

Reply:
So here it is checking the data field of the transaction to check if they were properly sent in case it were mallformed or something. Just an extra precaution